Reorganize HIP runtime API how-to (#3622)
ROCm-Developer-Tools/HIP: docs/how-to/hip_runtime_api/cooperative_groups.rst

Reviewer: I know this is probably out of scope, but I checked the whole file anyways: The deprecation note for multi-grid "(Deprecated since 5.0)" should be highlighted as an actual rst `.. note::`

Author: Fixed.

Author: Fixed.

Cooperative groups
*******************************************************************************

> Cooperative groups API is an extension to the HIP programming model, which

Reviewer: ```suggestion
The cooperative groups API is an extension to the HIP programming model, which
```

provides developers with a flexible, dynamic grouping mechanism for the
communicating threads. Cooperative groups let you define your own set of thread

> groups which may fit your user-cases better than those defined by the hardware.

Reviewer: ```suggestion
groups which may fit your use-cases better than those defined by the hardware.
```

This lets you specify the level of granularity for thread communication which
can lead to more efficient parallel decompositions.

The API is accessible in the ``cooperative_groups`` namespace after the 

> ``hip_cooperative_groups.h`` is included. The header contains the following

Reviewer: ```suggestion
``hip_cooperative_groups.h`` header is included. The header contains the following
```

* Get group properties member functions.

Cooperative groups thread model
================================================================================

> The thread hierarchy abstraction of cooperative groups are in :ref:`grid hierarchy <coop_thread_top_hierarchy>` and :ref:`block hierarchy <coop_thread_bottom_hierarchy>`.

Reviewer: ```suggestion
The thread hierarchy abstractions of cooperative groups are depicted in the following figures: :ref:`grid hierarchy <coop_thread_top_hierarchy>` and :ref:`block hierarchy <coop_thread_bottom_hierarchy>`.
```
It's even debatable, if the images should be linked here, as there is no corresponding text-anchor, they are on the same page even in the same section, and the caption doesn't use the same text that is used in the link
```suggestion
The thread hierarchy abstractions of cooperative groups are described in this section.
```
